hack/e2e.go

```diff
@@ -126,112 +127,106 @@ func main() {
+				log.Fatalf("error pushing e2e clsuter: %v", err)
 			}
 		}
 	}
 
-	upResources := ""
+	var upResources []byte
 	if *checkLeakedResources {
-		upResources = ListResources()
+		upResources, err = ListResources()
+		errs = appendError(errs, err)
 	}
 
-	success := true
-
 	if *ctlCmd != "" {
 		ctlArgs := strings.Fields(*ctlCmd)
 		os.Setenv("KUBE_CONFIG_FILE", "config-test.sh")
-		ctlSuccess := finishRunning("'kubectl "+*ctlCmd+"'", exec.Command("./cluster/kubectl.sh", ctlArgs...))
-		success = success && ctlSuccess
+		errs = appendError(errs, finishRunning("'kubectl "+*ctlCmd+"'", exec.Command("./cluster/kubectl.sh", ctlArgs...)))
 	}
 
 	if *upgradeArgs != "" {
-		upgradeSuccess := UpgradeTest(*upgradeArgs)
-		success = success && upgradeSuccess
+		errs = appendError(errs, UpgradeTest(*upgradeArgs))
 	}
 
 	if *test {
 		if *skewTests {
-			skewSuccess := SkewTest()
-			success = success && skewSuccess
+			errs = appendError(errs, SkewTest())
 		} else {
-			testSuccess := Test()
-			success = success && testSuccess
+			errs = appendError(errs, Test())
 		}
 	}
 
 	if *kubemark {
-		kubeSuccess := KubemarkTest()
-		success = success && kubeSuccess
-	}
-
-	if !success {
-		DumpClusterLogs(*dump)
+		errs = appendError(errs, KubemarkTest())
 	}
 
 	if *down {
-		tearSuccess := TearDown()
-		success = success && tearSuccess
+		if errs != nil && *dump != "" {
+			DumpClusterLogs(*dump)
+		}
+		errs = appendError(errs, TearDown())
 	}
 
 	if *checkLeakedResources {
 		log.Print("Sleeping for 30 seconds...") // Wait for eventually consistent listing
 		time.Sleep(30 * time.Second)
-		DiffResources(beforeResources, upResources, ListResources(), *dump)
-	}
-
-	if !success {
-		os.Exit(1)
+		afterResources, err := ListResources()
+		if err != nil {
+			errs = append(errs, err)
+		} else {
+			errs = appendError(errs, DiffResources(beforeResources, upResources, afterResources, *dump))
+		}
 	}
-}
 
-func writeOrDie(dir, name, data string) string {
-	f, err := os.Create(filepath.Join(dir, name))
-	if err != nil {
-		log.Fatal(err)
-	}
-	if _, err := f.WriteString(data); err != nil {
-		log.Fatal(err)
+	if len(errs) != 0 {
+		log.Fatalf("Encountered %d errors: %v", len(errs), errs)
 	}
-	if err := f.Close(); err != nil {
-		log.Fatal(err)
-	}
-	log.Printf("Created file: %s", f.Name())
-	return f.Name()
 }
 
-func DiffResources(before, clusterUp, after, location string) {
+func DiffResources(before, clusterUp, after []byte, location string) error {
 	if location == "" {
 		var err error
 		location, err = ioutil.TempDir("", "e2e-check-resources")
 		if err != nil {
-			log.Fatal(err)
+			return fmt.Errorf("Could not create e2e-check-resources temp dir: %s", err)
 		}
 	}
-	bp := writeOrDie(location, "gcp-resources-before.txt", before)
-	writeOrDie(location, "gcp-resources-cluster-up.txt", clusterUp)
-	ap := writeOrDie(location, "gcp-resources-after.txt", after)
+
+	var mode os.FileMode = 0664
+	bp := filepath.Join(location, "gcp-resources-before.txt")
```

Comment:
you have taken the Go unnecessarily-short-variable-name style to heart, I see. :)